presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -361,6 +375,18 @@ private static boolean isAggregationWithEmptyGroupBy(PlanNode planNode)
                 .orElse(false);
     }
 
+    private SubqueryExpression uncoercedSubquery(SubqueryExpression subquery)
```

Comment:
`TranslationMap.put` implicitly adds cast (based on `IdentityHashMap`) when adding `expression->symbol mapping`. I need to create copy of `SubqueryExpression`, so that there is no cast for it in `Analysis.coercions` map. After uncasted subquery is mapped to a symbol, casts are explicitly projected by `subqueryPlan.appendProjections`